src/vec.rs

```diff
@@ -846,6 +847,10 @@ mod tests {
     #[test]
     fn oom() {
         let mut vec: Vec<char> = Vec::new();
+        match FallibleVec::try_reserve(&mut vec, core::usize::MAX / std::mem::size_of::<char>()) {
+            Ok(_) => panic!("it should be OOM"),
+            _ => (),
+        }
```

Comment:
I recognize this matches the existing code, but it strikes me as odd to use a `match` here rather than a simple assert like
```suggestion
        assert!(FallibleVec::try_reserve(&mut vec, core::usize::MAX / std::mem::size_of::<char>()).is_err());
```

For the sake of clarity, it seems worth changing both to the more idiomatic approach.

Also, based on https://rust-lang.github.io/unsafe-code-guidelines/layout/scalars.html#isize-and-usize, should this be testing the tighter bound of `isize::MAX`?